pennylane/utils.py

```diff
@@ -417,3 +417,57 @@ def expand_vector(vector, original_wires, expanded_wires):
     )
 
     return qml.math.reshape(expanded_tensor, 2 ** M)
+
+
+def get_generator(op, return_matrix=False):
+    """Utility function to help return the generator of an operation.
+
+    This utility function is used to abstract away the API differences
+    between different operators that could be returned as
+    generators, including ``SparseHamiltonian``, ``Hamiltonian``,
+    ``Observable``, ``Tensor``, and ``Hermitian``.
+
+    This utility function should be removed once the aforementioned classes
+    no longer differ in behaviour.
+    """
+    gen = getattr(op, "generator", lambda: None)()
+
+    if gen is None:
+        raise ValueError(f"Operation {op.name} does not have a generator")
+
+    if op.num_params != 1:
+        raise ValueError(f"Operation {op.name} is not written in terms of a single parameter")
+
+    if isinstance(gen, (qml.Hermitian, qml.SparseHamiltonian)):
+        obs = gen
+        s = 1.0
+
+    elif isinstance(gen, qml.operation.Observable):
+        gen = 1.0 * gen  # convert to a qml.Hamiltonian
```

Comment:
Not 100% sure why you need to do that?

Reply:
This is a very sneaky way of taking something that is in the set `{Observable, Tensor, Hamiltonian}`, and ensure it becomes only a `Hamiltonian` lol.

The reason I need to do this here is because only the Hamiltonian currently defines `ops` and `coeffs`.

However, once `terms` etc. is added, I think this can be removed; all operators can be identically queried for their terms (or, if they do not have terms, grabbing their matrix)